module/extension/Director/src/director/run_task_pack.cpp

```diff
@@ -181,6 +181,51 @@ namespace module::extension {
             }
         }
 
+        // Check if a Wait command was emitted and schedule to run the Provider again
+        //
+        for (const auto& t : requested_tasks) {
+            if (t->type == typeid(::extension::behaviour::Wait)) {
+                if (requested_tasks.size() > 1) {
+                    log<WARN>("Wait task was emitted with other tasks, the other tasks will be ignored");
+                }
+
+                // Schedule the Provider to run again
+                // Get the time to wait for
+                auto wait_data                 = std::static_pointer_cast<::extension::behaviour::Wait>(t->data);
+                std::chrono::nanoseconds delay = std::chrono::nanoseconds(wait_data->time - NUClear::clock::now());
+
+                // If the delay is over, just run the provider
+                if (delay <= std::chrono::nanoseconds(0)) {
+                    run_task_on_provider(group.active_task, provider, RunReason::WAIT);
+                    return;
+                }
+
+                // Otherwise, send it to the ChronoController to handle
+                // Make a weak pointer to the task so we can check if it still exists when the task is run
+                std::weak_ptr<component::DirectorTask> weak_task = t;
+                emit(std::make_unique<NUClear::dsl::operation::ChronoTask>(
+                    [this, provider, weak_task](const NUClear::clock::time_point&) {
+                        // Check if the task still exists
+                        if (weak_task.expired()) {
+                            return false;
+                        }
+
+                        // Get the task
+                        auto task = weak_task.lock();
+                        emit(std::make_unique<WaitDelay>(provider));
```

Comment:
Better to just auto task = weak_task.lock() and then do the emit if(task);

Technically the task could expire between you calling expired and locking
```suggestion
                        // Check if the task still exists
                        auto task = weak_task.lock();
                        if (weak_task) {
                            emit(std::make_unique<WaitDelay>(provider));
                        }
```